python/lsst/fgcmcal/fgcmOutputIlluminationCorrection.py

```diff
@@ -0,0 +1,348 @@
+            raise ValueError("cycleNumber must be of integer format")
+
+        if not config.use_flat_metadata:
+            del self.flat
+        else:
+            def lookup_flat(dataset_type, registry, data_id, collections):
+
+                time = Time(config.epoch_time, format=config.epoch_format)
+
+                return [
+                    registry.findDataset(
+                        dataset_type,
+                        data_id,
+                        physical_filter=physical_filter,
+                        timespan=Timespan(
+                            time - TimeDelta(1, format="sec"),
+                            time + TimeDelta(1, format="sec"),
+                        ),
+                        collections=collections,
+                    )
+                    for physical_filter in config.physical_filters
+                ]
+
+            self.flat = dataclasses.replace(self.flat, lookupFunction=lookup_flat)
+
+
+class FgcmOutputIlluminationCorrectionConfig(
+    PipelineTaskConfig,
+    pipelineConnections=FgcmOutputIlluminationCorrectionConnections,
+):
+    """Configuration for FgcmOutputIlluminationCorrectionTask."""
+
+    use_flat_metadata = lsst.pex.config.Field(
+        doc="Use flat-field metadata for illumination correction metadata?",
+        dtype=bool,
+        default=True,
+    )
+    epoch_time = lsst.pex.config.Field(
+        doc="Time string (UTC) that corresponds to a date in the desired epoch.",
+        dtype=str,
+        default=None,
+        optional=False,
+    )
+    epoch_format = lsst.pex.config.Field(
+        doc="Format for time string (e.g. iso, mjd, etc.), used by astropy.time.Time()",
+        dtype=str,
+        default="iso",
+    )
+    physical_filters = lsst.pex.config.ListField(
+        doc="List of physical filters to produce illumination corrections.",
+        dtype=str,
+        default=[],
+    )
+    include_wcs_jacobian = lsst.pex.config.Field(
+        doc="Include WCS jacobian in illumination correction?",
+        dtype=bool,
+        default=True,
+    )
+    approximate_wcs_jacobian = lsst.pex.config.Field(
+        doc="Use a Chebyshev approximation of the WCS jacobian in illumination correction?",
+        dtype=bool,
+        default=True,
+    )
+
+    def validate(self):
+        try:
+            _ = Time(self.epoch_time, format=self.epoch_format)
+        except Exception as e:
+            raise ValueError(
+                "Could not parse epoch_time/epoch_format ", e)
+
+
+class FgcmOutputIlluminationCorrectionTask(PipelineTask):
+    """Output illumination corrections from fgcm."""
+    ConfigClass = FgcmOutputIlluminationCorrectionConfig
+    _DefaultName = "fgcmOutputIlluminationCorrection"
+
+    def runQuantum(self, butlerQC, inputRefs, outputRefs):
+
+        # Get the flat metadata if necessary.
+        # Get the first exposure if we can to select the epoch.
+        # Or select the epoch number.  These are handed to the task.
```

Comment:
Is comment still fully applicable?  I don't see any epoch numbers getting passed to the task.